operators/pkg/forge/virtualmachines.go

```diff
@@ -222,17 +222,29 @@ func VirtualMachineReadinessProbe(environment *clv1alpha2.Environment) *virtv1.P
 	}
 }
 
+// DataVolumeSourceForge forges the DataVolumeSource for DataVolumeTemplate.
+func DataVolumeSourceForge(environment *clv1alpha2.Environment) cdiv1beta1.DataVolumeSource {
+	if environment.EnvironmentType == clv1alpha2.ClassCloudVM {
+		return cdiv1beta1.DataVolumeSource{
+			HTTP: &cdiv1beta1.DataVolumeSourceHTTP{
+				URL: environment.Image,
+			},
+		}
+	}
+	return cdiv1beta1.DataVolumeSource{
+		Registry: &cdiv1beta1.DataVolumeSourceRegistry{
+			URL:       urlDockerPrefix + environment.Image,
+			SecretRef: cdiSecretName,
+		},
+	}
+}
```

Comment:
Should add a test to validate the correctness of this function, by adding the appropriate `When` cases to the `"The DataVolumeTemplate is forged"` Context block.